disnake/flags.py

```diff
@@ -2892,3 +2934,152 @@ def bot_dm(self):
     def private_channel(self):
         """:class:`bool`: Returns ``True`` if the command is usable in DMs and group DMs with other users."""
         return 1 << 2
+
+
+class EmbedFlags(BaseFlags):
+    """Wraps up Discord Embed flags.
+
+    .. collapse:: operations
+
+        .. describe:: x == y
+
+            Checks if two EmbedFlags instances are equal.
+        .. describe:: x != y
+
+            Checks if two EmbedFlags instances are not equal.
+        .. describe:: x <= y
+
+            Checks if an EmbedFlags instance is a subset of another EmbedFlags instance.
+        .. describe:: x >= y
+
+            Checks if an EmbedFlags instance is a superset of another EmbedFlags instance.
+        .. describe:: x < y
+
+            Checks if an EmbedFlags instance is a strict subset of another EmbedFlags instance.
+        .. describe:: x > y
+
+            Checks if an EmbedFlags instance is a strict superset of another EmbedFlags instance.
+        .. describe:: x | y, x |= y
+
+            Returns a new EmbedFlags instance with all enabled flags from both x and y.
+            (Using ``|=`` will update in place).
+        .. describe:: x & y, x &= y
+
+            Returns a new EmbedFlags instance with only flags enabled on both x and y.
+            (Using ``&=`` will update in place).
+        .. describe:: x ^ y, x ^= y
+
+            Returns a new EmbedFlags instance with only flags enabled on one of x or y, but not both.
+            (Using ``^=`` will update in place).
+        .. describe:: ~x
+
+            Returns a new EmbedFlags instance with all flags from x inverted.
+        .. describe:: hash(x)
+
+            Returns the flag's hash.
+        .. describe:: iter(x)
+
+            Returns an iterator of ``(name, value)`` pairs. This allows it
+            to be, for example, constructed as a dict or a list of pairs.
+            Note that aliases are not shown.
+
+        Additionally supported are a few operations on class attributes.
+
+        .. describe:: EmbedFlags.y | EmbedFlags.z, EmbedFlags(y=True) | EmbedFlags.z
+
+            Returns an EmbedFlags instance with all provided flags enabled.
+
+        .. describe:: ~EmbedFlags.y
+
+            Returns an EmbedFlags instance with all flags except ``y`` inverted from their default value.
+
+    .. versionadded:: 2.11
+
+    Attributes
+    ----------
+    value: :class:`int`
+        The raw value. You should query flags via the properties
+        rather than using this raw value.
+    """
+
+    @flag_value
+    def contains_explicit_media(self):
```

Comment:
Thoughts on calling this something different? Like `is_nsfw` or `is_explicit`

Reply:
I think following the API naming is okay